internal/eval/evalers_test.go

```diff
@@ -1448,6 +1448,175 @@ func TestHasNode(t *testing.T) {
 	}
 }
 
+func TestGetTagNode(t *testing.T) {
+	t.Parallel()
+
+	const (
+		knownType   = types.EntityType("knownType")
+		unknownType = types.EntityType("")
+		knownID     = "knownID"
+		unknownID   = ""
+		knownTag    = types.String("knownTag")
+		unknownTag  = types.String("unknownTag")
+		knownAttr   = "knownAttr"
+		notString   = types.Long(42)
+		value       = types.Long(42)
+	)
+
+	tests := []struct {
+		name   string
+		lhs    Evaler
+		rhs    Evaler
+		result types.Value
+		err    error
+	}{
+		{"ObjectTypeError",
+			newLiteralEval(types.True),
+			newLiteralEval(knownTag),
+			zeroValue(),
+			ErrType},
+		{"SubjectTypeError",
+			newLiteralEval(types.NewEntityUID(knownType, knownID)),
+			newLiteralEval(notString),
+			zeroValue(),
+			ErrType},
+		{"TagOnRecord",
+			newLiteralEval(types.NewRecord(nil)),
+			newLiteralEval(knownTag), zeroValue(),
+			ErrType},
+		{"ProgrammaticTag",
+			newLiteralEval(types.NewEntityUID(knownType, knownID)),
+			newAttributeAccessEval(newLiteralEval(types.NewEntityUID(knownType, knownID)), knownAttr),
+			value,
+			nil,
+		},
+		{"KnownTag",
+			newLiteralEval(types.NewEntityUID(knownType, knownID)),
+			newLiteralEval(knownTag),
+			value,
+			nil},
+		{"UnknownTag",
+			newLiteralEval(types.NewEntityUID(knownType, knownID)),
+			newLiteralEval(unknownTag),
+			zeroValue(),
+			errTagAccess},
+		{"UnknownEntity",
+			newLiteralEval(types.NewEntityUID("unknownType", unknownID)),
+			newLiteralEval(knownTag),
+			zeroValue(),
+			errEntityNotExist},
+		{"UnspecifiedEntity",
+			newLiteralEval(types.NewEntityUID("", "")),
```

Comment:
Yeah, this totally shouldn't be allowed.

Reply:
I've renamed this to `ZeroEntity`—while it's not a possible call path, we do need this test to maintain 100% net new code coverage. Done here: https://github.com/cedar-policy/cedar-go/pull/77/commits/9074d5b00f44be4346c13ac8b7270a09b06c541a.